client/lib/analytics/super-props.js

```diff
@@ -23,9 +30,10 @@ const getSuperProps = ( reduxStore ) => ( eventProperties ) => {
 	}
 
 	const path = eventProperties.path ?? getCurrentRoute( state );
-
 	const omitSelectedSite = ! eventProperties.force_site_id && shouldReportOmitBlogId( path );
-	const selectedSite = omitSelectedSite ? null : getSelectedSite( state );
+	const selectedSite = omitSelectedSite
+		? null
+		: getSelectedSite( state ) || getSite( state, getSiteSlugOrIdFromURLSearchParams() );
```

Comment:
It might be better to set the selected site ID first if possible. I'm not sure whether relying on the query string inside this function is good or not. But we can address this later.

Reply:
WDYT about using [getPrimarySiteId()](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/selectors/get-primary-site-id.js)?

Reply:
I guess the value is not the same, right 🤔

Reply:
Yeah, might not be the same. What did you mean but set the selected site ID?

Reply:
I mean dispatching the `setSelectedSiteId` when we fetched the site in the stepper framework so the `getSelectedSite` selector can get the correct site and we don't need to fall back to the `getSite( state, getSiteSlugOrIdFromURLSearchParams() )`

Reply:
Yes, I think we can address this later when needed since at the moment Stepper heavily relies on the query string parameters for fetching site slug/ID.